src/ext/framework/mod.rs

```diff
@@ -194,6 +197,30 @@ impl Framework {
 
     #[doc(hidden)]
     pub fn dispatch(&mut self, context: Context, message: Message) {
+        match self.configuration.account_type {
+            AccountType::Selfbot => {
+                if message.author.id != CACHE.read().unwrap().user.id {
+                    return;
+                }
+            },
+            AccountType::Bot => {
+                if message.author.bot {
+                    return;
+                }
+            },
+            AccountType::Any => {},
+            AccountType::Automatic => {
+                if CACHE.read().unwrap().user.bot {
```

Comment:
The latter condition (in the else) will unlock the cache a second time, if you bind `CACHE.read().unwrap()` to `cache` and then access it via `cache.user.bot` and `cache.user.id`, then it'll only need to be unlocked once.